account/tests.py

```diff
@@ -0,0 +1,155 @@
+from django.contrib.auth.models import User
+from .models.profile import Profile, UserType
+from .models.professional import Professions, Professional
+from .models.client import Client
+import datetime
+import pytest
+
+
+USERNAME = 'testusername'
+PASSWORD = 'testpassword'
+FIRST_NAME = 'testfirstname'
+LAST_NAME = 'testlastname'
+EMAIL = 'test@test.test'
+LAST_LOGIN = datetime.datetime.now()
+
+PROFESSIONAL_TYPE = UserType.Professional
+CLIENT_TYPE = UserType.Client
+PHONE_NUMBER = '1234567890'
+COUNTRY = 'testcountry'
+CITY = 'testcity'
+ADDRESS = 'testaddress'
+
+PROFESSION = Professions.Handyman
+DESCRIPTION = 'test description'
+
+BIRTHDAY = datetime.date(2000, 1, 1)
+
+
+@pytest.fixture
+def user():
+    user = User.objects.create_user(
+        username=USERNAME,
+        password=PASSWORD,
+        first_name=FIRST_NAME,
+        last_name=LAST_NAME,
+        email=EMAIL,
+        last_login=LAST_LOGIN
+    )
+    return user
+
+
+@pytest.fixture
+def profile(user):
+    profile = Profile.objects.create(
+        user_id=user,
+        user_type=PROFESSIONAL_TYPE,
+        phone_number=PHONE_NUMBER,
+        country=COUNTRY,
+        city=CITY,
+        address=ADDRESS
+    )
+    return profile
+
+
+@pytest.fixture
+def professional(profile):
+    professional_profile = profile
+    professional_profile.user_type = PROFESSIONAL_TYPE,
+    professional = Professional.objects.create(
+        profile_id=professional_profile,
+        profession=PROFESSION,
+        description=DESCRIPTION
+    )
+    return professional
+
+
+@pytest.fixture
+def client(profile):
+    client_profile = profile
+    client_profile.user_type = CLIENT_TYPE,
+    client = Client.objects.create(
+        profile_id=profile,
+        birthday=BIRTHDAY
+    )
+    return client
+
+
+@pytest.mark.django_db
+class TestProfileModel:
```

Comment:
What happens to the Client/Pro models that are connected to this profile?

Did you think to add delete_profile method, that remove the Clien/Pro or User?  

Those are the interesting tests?

Reply:
I cant add a new Pro/Client if I dont have a Profile that attached to them
and for that I have the test_delete_user_deletes_client/test_delete_user_deletes_professional functions.